frontend/src/container/MetricsExplorer/Explorer/Explorer.tsx

```diff
@@ -48,7 +80,64 @@ function Explorer(): JSX.Element {
 	const [showOneChartPerQuery, toggleShowOneChartPerQuery] = useState(
 		isOneChartPerQueryEnabled,
 	);
+	const [disableOneChartPerQuery, toggleDisableOneChartPerQuery] = useState(
+		false,
+	);
 	const [selectedTab] = useState<ExplorerTabs>(ExplorerTabs.TIME_SERIES);
+	const [yAxisUnit, setYAxisUnit] = useState<string | undefined>();
+
+	useEffect(() => {
+		// Set the y axis unit to the first metric unit if
+		// 1. There is one metric unit and it is not empty
+		// 2. All metric units are the same and not empty
+		// Else, set the y axis unit to empty if
+		// 1. There are more than one metric units and they are not the same
+		// 2. There are no metric units
+		// 3. There is exactly one metric unit but it is empty/undefined
+		if (units.length === 0) {
+			setYAxisUnit(undefined);
+		} else if (units.length === 1 && units[0]) {
+			setYAxisUnit(units[0]);
+		} else if (units.length === 1 && !units[0]) {
+			setYAxisUnit(undefined);
+		} else if (areAllMetricUnitsSame) {
+			if (units[0]) {
+				setYAxisUnit(units[0]);
+			} else {
+				setYAxisUnit(undefined);
+			}
+		} else if (units.length > 1 && !areAllMetricUnitsSame) {
+			setYAxisUnit(undefined);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [JSON.stringify(units), areAllMetricUnitsSame]);
```

Comment:
stringify is a code smell here

Reply:
Passing the units array in itself was causing undesired re-renders. So. I tried to stringify it. But I'll clean up the dependency array here in a better way.